### ISSUE TITLE
Issue running playbook to install controller and nodes

### ISSUE THREAD

**Issue by distsystemsarchitect** [open-ness/specs#51]
While trying to run the deploy scripts (deploy_ne.sh), I am getting errors of version compatibility of python.  This is on a clean install of Centos on KVM.  Full connectivity between controller node and edge node (tested with ansible -m ping all command).

I am getting the following error:
STDOUT:

  File: ‘/opt/edgecontroller’
  Size: 4096      	Blocks: 8          IO Block: 4096   directory
Device: fd00h/64768d	Inode: 50761770    Links: 27
Access: (0755/drwxr-xr-x)  Uid: (    0/    root)   Gid: (    0/    root)
Context: unconfined_u:object_r:usr_t:s0
Access: 2020-08-24 09:17:56.527509889 -0700
Modify: 2020-08-24 09:17:57.863589833 -0700
Change: 2020-08-24 09:17:57.863589833 -0700
 Birth: -

TASK [openness/onprem/master : remove all Docker containers] ******************************************************************************************************************************************************************************
task path: /root/openness/openness-experience-kits/roles/openness/onprem/master/tasks/subtasks/remove_docker_containers.yml:19
fatal: [controller]: FAILED! => {
    "changed": true,
    "cmd": [
        "docker-compose",
        "rm",
        "--stop",
        "-f"
    ],
    "delta": "0:00:00.192591",
    "end": "2020-08-24 09:18:23.738305",
    "rc": 1,
    "start": "2020-08-24 09:18:23.545714"
}

STDERR:

/usr/lib64/python2.7/site-packages/cryptography/__init__.py:39: CryptographyDeprecationWarning: Python 2 is no longer supported by the Python core team. Support for it is now deprecated in cryptography, and will be removed in a future release.
  CryptographyDeprecationWarning,
Traceback (most recent call last):
  File "/usr/bin/docker-compose", line 7, in <module>
    from compose.cli.main import main
  File "/usr/lib/python2.7/site-packages/compose/cli/main.py", line 17, in <module>
    import docker
  File "/usr/lib/python2.7/site-packages/docker/__init__.py", line 2, in <module>
    from .api import APIClient
  File "/usr/lib/python2.7/site-packages/docker/api/__init__.py", line 2, in <module>
    from .client import APIClient
  File "/usr/lib/python2.7/site-packages/docker/api/client.py", line 10, in <module>
    from .build import BuildApiMixin
  File "/usr/lib/python2.7/site-packages/docker/api/build.py", line 6, in <module>
    from .. import auth
  File "/usr/lib/python2.7/site-packages/docker/auth.py", line 9, in <module>
    from .utils import config
  File "/usr/lib/python2.7/site-packages/docker/utils/__init__.py", line 3, in <module>
    from .decorators import check_resource, minimum_version, update_headers
  File "/usr/lib/python2.7/site-packages/docker/utils/decorators.py", line 4, in <module>
    from . import utils
  File "/usr/lib/python2.7/site-packages/docker/utils/utils.py", line 13, in <module>
    from .. import tls
  File "/usr/lib/python2.7/site-packages/docker/tls.py", line 5, in <module>
    from .transport import SSLHTTPAdapter
  File "/usr/lib/python2.7/site-packages/docker/transport/__init__.py", line 11, in <module>
    from .sshconn import SSHHTTPAdapter
  File "/usr/lib/python2.7/site-packages/docker/transport/sshconn.py", line 1, in <module>
    import paramiko
  File "/usr/lib/python2.7/site-packages/paramiko/__init__.py", line 22, in <module>
    from paramiko.transport import SecurityOptions, Transport
  File "/usr/lib/python2.7/site-packages/paramiko/transport.py", line 89, in <module>
    from paramiko.dsskey import DSSKey
  File "/usr/lib/python2.7/site-packages/paramiko/dsskey.py", line 25, in <module>
    from cryptography.hazmat.primitives import hashes, serialization
  File "/usr/lib64/python2.7/site-packages/cryptography/hazmat/primitives/serialization/__init__.py", line 22, in <module>
    from cryptography.hazmat.primitives.serialization.ssh import (
  File "/usr/lib64/python2.7/site-packages/cryptography/hazmat/primitives/serialization/ssh.py", line 27, in <module>
    from bcrypt import kdf as _bcrypt_kdf
  File "/usr/lib64/python2.7/site-packages/bcrypt/__init__.py", line 57
    def gensalt(rounds: int = 12, prefix: bytes = b"2b") -> bytes:
                      ^
SyntaxError: invalid syntax

MSG:

non-zero return code

PLAY RECAP ********************************************************************************************************************************************************************************************************************************
controller                 : ok=67   changed=23   unreachable=0    failed=1    skipped=37   rescued=0    ignored=0

------------------[ My version of CentOS ]--------------------------
controller  root ~/openness/openness-experience-kits:> hostnamectl
   Static hostname: controller
         Icon name: computer-vm
           Chassis: vm
        Machine ID: ........2860dd8b4c6d97a0a5ef4...........
           Boot ID: .......4bf219a408699bf83b27.............
    Virtualization: kvm
  Operating System: CentOS Linux 7 (Core)
       CPE OS Name: cpe:/o:centos:centos:7
            Kernel: Linux 3.10.0-957.el7.x86_64
      Architecture: x86-64

------------------[ My inventory file ]--------------------------
# SPDX-License-Identifier: Apache-2.0
# Copyright (c) 2019-2020 Intel Corporation

[all]
controller ansible_ssh_user=root ansible_host=192.168.122.113
edge ansible_ssh_user=root ansible_host=192.168.122.22

[controller_group]
controller

[edgenode_group]
edge

[edgenode_vca_group]

[ptp_master]
controller

[ptp_slave_group]
edge

------------------[ Additional effects ]--------------------------

Additionally, this renders my ansible installation unusable, even if I remove and install again (yum remove / install ).  I have tried updating python to later versions, but then get other syntax errors).

controller  root ~:> ansible
/usr/lib64/python2.7/site-packages/cryptography/__init__.py:39: CryptographyDeprecationWarning: Python 2 is no longer supported by the Python core team. Support for it is now deprecated in cryptography, and will be removed in a future release.
  CryptographyDeprecationWarning,
ERROR! Unexpected Exception, this is probably a bug: invalid syntax (__init__.py, line 57)
the full traceback was:

Traceback (most recent call last):
  File "/bin/ansible", line 92, in <module>
    mycli = getattr(__import__("ansible.cli.%s" % sub, fromlist=[myclass]), myclass)
  File "/usr/lib/python2.7/site-packages/ansible/cli/__init__.py", line 22, in <module>
    from ansible.inventory.manager import InventoryManager
  File "/usr/lib/python2.7/site-packages/ansible/inventory/manager.py", line 38, in <module>
    from ansible.plugins.loader import inventory_loader
  File "/usr/lib/python2.7/site-packages/ansible/plugins/loader.py", line 23, in <module>
    from ansible.parsing.utils.yaml import from_yaml
  File "/usr/lib/python2.7/site-packages/ansible/parsing/utils/yaml.py", line 17, in <module>
    from ansible.parsing.yaml.loader import AnsibleLoader
  File "/usr/lib/python2.7/site-packages/ansible/parsing/yaml/loader.py", line 30, in <module>
    from ansible.parsing.yaml.constructor import AnsibleConstructor
  File "/usr/lib/python2.7/site-packages/ansible/parsing/yaml/constructor.py", line 30, in <module>
    from ansible.parsing.vault import VaultLib
  File "/usr/lib/python2.7/site-packages/ansible/parsing/vault/__init__.py", line 52, in <module>
    CRYPTOGRAPHY_BACKEND = default_backend()
  File "/usr/lib64/python2.7/site-packages/cryptography/hazmat/backends/__init__.py", line 15, in default_backend
    from cryptography.hazmat.backends.openssl.backend import backend
  File "/usr/lib64/python2.7/site-packages/cryptography/hazmat/backends/openssl/__init__.py", line 7, in <module>
    from cryptography.hazmat.backends.openssl.backend import backend
  File "/usr/lib64/python2.7/site-packages/cryptography/hazmat/backends/openssl/backend.py", line 16, in <module>
    from cryptography import utils, x509
  File "/usr/lib64/python2.7/site-packages/cryptography/x509/__init__.py", line 8, in <module>
    from cryptography.x509.base import (
  File "/usr/lib64/python2.7/site-packages/cryptography/x509/base.py", line 22, in <module>
    from cryptography.x509.extensions import Extension, ExtensionType
  File "/usr/lib64/python2.7/site-packages/cryptography/x509/extensions.py", line 22, in <module>
    from cryptography.hazmat.primitives import constant_time, serialization
  File "/usr/lib64/python2.7/site-packages/cryptography/hazmat/primitives/serialization/__init__.py", line 22, in <module>
    from cryptography.hazmat.primitives.serialization.ssh import (
  File "/usr/lib64/python2.7/site-packages/cryptography/hazmat/primitives/serialization/ssh.py", line 27, in <module>
    from bcrypt import kdf as _bcrypt_kdf
  File "/usr/lib64/python2.7/site-packages/bcrypt/__init__.py", line 57
    def gensalt(rounds: int = 12, prefix: bytes = b"2b") -> bytes:
                      ^
SyntaxError: invalid syntax

Any help would be greatly appreciated.

Thanks in advance!

**Comment by archie951** [comment]
Hi @distsystemsarchitect ,
The error seems to be because of some updated cryptography package here. I suggest you uninstall it using below command on both controller and node.
**pip uninstall -y cryptography**

Run the deploy script after removing it.
If you face this error again, please install watchdog on both node and controller and run the below command parallelly while running the deploy_ne.sh script.
 watch pip uninstall -y cryptography

Basically this command will be executed every 2 seconds, so if by any chance cryptography get installed anytime during the installation process, using watchdog will keep on clearing it automatically without manual intervention. Please let us know if this works.

Regards,
Archit

**Comment by archie951** [comment]
Hi @distsystemsarchitect, 

Are you able to install OpenNESS 20.06 successfully. If you are facing any issues please let us know.

Regards,
Archit

**Comment by archi-wipro** [comment]
Hi @distsystemsarchitect ,

The permanent fix for this issue has been solved by the OpenNESS 20.06.01 patch release. Please clone the repository again and you won't face this issue.

@amr-mokhtar : Please close this ticket. Thanks!

Regards,
Archit

**Comment by archie951** [comment]
Hi @distsystemsarchitect ,

The permanent fix for this issue has been solved by the OpenNESS 20.06.01 patch release. Please clone the repository again and you won't face this issue.

@amr-mokhtar : Please close this ticket. Thanks!

Regards,
Archit

**Comment by sushantsj** [comment]
I am also facing the same issue and tried uninstalling cryptography as mentioned above but uninstalling it on controller and node, it is showing that cryptography is not installed. Can you help?

**Comment by archie951** [comment]
Hi @sushantsj,

This issue has now been resolved by the new OpenNESS 20.06.01 patch release. Could you please try doing a fresh installation once and let us know if you face the same issue. Try doing a cleanup and clone the openness-experience-kits once again 

Thanks,
Archit

**Comment by sushantsj** [comment]
@archie951  
Facing this issue while ./deploy_ne.sh node01
Error: TASK [machine_setup/os_setup : Install IUS Repository]
Failure Downloading https://repo.ius.io/ius-release-e17.rpm, Request Failed:<url open error [Errno-2] Name or service not known>
Any help would be appreciated

**Comment by sushantsj** [comment]
> Hi @sushantsj,
> 
> This issue has now been resolved by the new OpenNESS 20.06.01 patch release. Could you please try doing a fresh installation once and let us know if you face the same issue. Try doing a cleanup and clone the openness-experience-kits once again
> 
> Thanks,
> Archit

It is the latest release and the issue is not in node but in controller
1. ./deploy_ne.sh controller //failing related to python issue mentioned above.
2. ./deploy_ne.sh node // failing while pulling IUS repository i.e Failure Downloading https://repo.ius.io/ius-release-e17.rpm, Request Failed:<url open error [Errno-2] Name or service not known>

**Comment by sushantsj** [comment]
Hi @archie951 ,
Cloned the latest release, still facing the same issue.
Can you help?

**Comment by archie951** [comment]
Hi @sushantsj, 
Please note that there is no repository by the name of  https://repo.ius.io/ius-release-e17.rpm. It is rather present in roles/machine_setup/os_setup/tasks/install_base_os_packages.yml as  **https://repo.ius.io/ius-release-el7.rpm** .
Can you please check the value set in above yml. I hope you have not made any changes in the openness experience kits and only have changed hugepages according to your system requirement. Rest of the packages and dependencies will be installed by OpenNESS itself. 
If you still face issues please share with us the log files for debugging.

Thanks and Regards,
Archit Agarwal

**Comment by sushantsj** [comment]
Hi Archit,
Error while running ./deploy_ne.sh controller


Error: fatal: [controller]: FAILED! => {
    "changed": false
}

MSG:

Error building ovs-dpdk - code: None, message: Error processing tar
file(exit status 1): write /root/ovs-2.12.0-5/ovsdb/ovsdb-server: no space
left on device, logs: [u'Step 1/17 : FROM centos:7', u'\n', u' --->
7e6257c9f8d8\n', u'Step 2/17 : ENV PYTHONDONTWRITEBYTECODE yes', u'\n', u'
---> Using cache\n', u' ---> b7fa4bb40714\n', u'Step 3/17 : RUN yum install
-y          gcc gcc-c++ make autoconf automake libtool rpm-build
PyYAML bind-utils         openssl         numactl-libs numactl-devel
  firewalld-filesystem         libpcap         hostname         iproute
strace socat nc         unbound unbound-devel         libpcap-devel
libmnl-devel         libibumad         libibverbs-devel         libibverbs
        libmlx5         libibverbs-utils         dpdk-devel', u'\n', u'
---> Using cache\n', u' ---> 1fa7c29eca49\n', u'Step 4/17 : ENV
OVS_VERSION=2.12.0', u'\n', u' ---> Using cache\n', u' --->
7245182bf793\n', u'Step 5/17 : ENV OVS_SUBVERSION=5', u'\n', u' ---> Using
cache\n', u' ---> f76622f9f65c\n', u'Step 6/17 : ENV DPDK_VERSION=18.11.6',
u'\n', u' ---> Using cache\n', u' ---> dd42533850da\n', u'Step 7/17 : ENV
DPDK_DIR=/opt/dpdk-$DPDK_VERSION', u'\n', u' ---> Using cache\n', u' --->
977ce8e9240f\n', u'Step 8/17 : ENV DPDK_TARGET=x86_64-native-linuxapp-gcc',
u'\n', u' ---> Using cache\n', u' ---> 615fece946b8\n', u'Step 9/17 : ENV
DPDK_BUILD=$DPDK_DIR/$DPDK_TARGET', u'\n', u' ---> Using cache\n', u' --->
6a6f9cd61213\n', u'Step 10/17 : COPY . $DPDK_DIR', u'\n', u' ---> Using
cache\n', u' ---> 59c3da31b0ee\n', u'Step 11/17 : RUN cd ~ &&     curl -OL
https://github.com/alauda/ovs/archive/$OVS_VERSION-$OVS_SUBVERSION.tar.gz
&&     tar xf $OVS_VERSION-$OVS_SUBVERSION.tar.gz &&     rm -f
$OVS_VERSION-$OVS_SUBVERSION.tar.gz &&     cd
ovs-$OVS_VERSION-$OVS_SUBVERSION &&     sed -e \'s/@VERSION@/0.0.1/\' rhel/
openvswitch-fedora.spec.in > /tmp/tmp_ovs.spec &&     yum-builddep -y
/tmp/tmp_ovs.spec &&     ./boot.sh &&     ./configure --prefix=/usr/
--localstatedir=/var --sysconfdir=/etc --with-dpdk=$DPDK_BUILD &&     make
-j$(nproc) && make rpm-fedora RPMBUILD_OPT="--with dpdk --without check" &&
    make install &&    rpm -ivh --nodeps
https://github.com/alauda/ovs/releases/download/${OVS_VERSION}-${OVS_SUBVERSION}/ovn-${OVS_VERSION}-${OVS_SUBVERSION}.el7.x86_64.rpm
&&     rpm -ivh
https://github.com/alauda/ovs/releases/download/${OVS_VERSION}-${OVS_SUBVERSION}/ovn-vtep-${OVS_VERSION}-${OVS_SUBVERSION}.el7.x86_64.rpm
&&     rpm -ivh
https://github.com/alauda/ovs/releases/download/${OVS_VERSION}-${OVS_SUBVERSION}/ovn-central-${OVS_VERSION}-${OVS_SUBVERSION}.el7.x86_64.rpm
&&     rpm -ivh
https://github.com/alauda/ovs/releases/download/${OVS_VERSION}-${OVS_SUBVERSION}/ovn-host-${OVS_VERSION}-${OVS_SUBVERSION}.el7.x86_64.rpm',
u'\n', u' ---> Running in 7870a3df54e8\n', u'\x1b[91m  % Total    %
Received % Xferd  Average Speed   Time    Time     Time  Current\n


Thanks,
Sushant

**Comment by archie951** [comment]
Hi @sushantsj, 

Can you please let us know your controller and node hardware specifics. How much HDD and hard-disk space have been allocated and how much is left by doing df -h and free -h?

Regards,
Archit

**Comment by sushantsj** [comment]
Hi Archit,

Controller:
OS: RedHat(64 bit) RAM: 4 GB, HDD: 8 GB, Processors: 4
Node:
OS: RedHat(64 bit), RAM: 3 GB, HDD: 8GB, Processors: 4

Could not run 1. df -h and free-f because after running ./deploy_ne.sh
controller and for Node the virtual machines are getting corrupt, so have
to restore it again.

Thank you,
Sushant Jha

**Comment by archie951** [comment]
Hi @sushantsj ,

The controller and node created by you are too small for OpenNESS deployment. That is the reason it is getting corrupt. Atleast 30GB or more HDD is required. Also the RAM is too less. The huge pages will never be sufficient enough. I would suggest to use atleast 16GB RAM for controller and 32GB for node. 
Please refer the documentation.
https://github.com/open-ness/specs/blob/master/doc/getting-started/network-edge/controller-edge-node-setup.md#qa

Regards,
Archit

**Comment by sushantsj** [comment]
Hey,
Thank you, will change the configuration and try again. Will let you
know how it goes.

Thank you,
Sushant

**Comment by sushantsj** [comment]
Hi archie,
Setup is done but I am facing this issue now, Can you have a look on it?
TASK [telemetry/tas : build TAS]
***************************************************************************************************************
task path:
/root/openness-experience-kits/roles/telemetry/tas/tasks/main.yml:154
fatal: [controller]: FAILED! => {
    "changed": true,
    "cmd": "source /etc/profile && make build",
    "delta": "0:00:50.821848",
    "end": "2020-09-18 13:11:16.200320",
    "rc": 2,
    "start": "2020-09-18 13:10:25.378472"
}

STDOUT:

CGO_ENABLED=0 GO111MODULE=on go build -ldflags="-s -w" -o ./bin/controller
./cmd/tas-policy-controller


STDERR:

go: finding k8s.io/client-go v0.0.0-20191016110837-54936ba21026
go: finding k8s.io/api v0.0.0-20191016110246-af539daaa43a
go: finding k8s.io/apiextensions-apiserver
v0.0.0-20190918201827-3de75813f604
go: finding k8s.io/apimachinery v0.0.0-20191004115701-31ade1b30762
go: finding k8s.io/metrics v0.0.0-20190918202012-3c1ca76f5bda
go: finding k8s.io/klog v0.3.1
go: finding golang.org/x/net v0.0.0-20190812203447-cdfb69ac37fc
go: finding golang.org/x/sys v0.0.0-20190312061237-fead79001313
go: finding golang.org/x/sys v0.0.0-20180909124046-d0be0721c37e
go: finding k8s.io/client-go v0.0.0-20190918200256-06eb1244587a
go: finding k8s.io/api v0.0.0-20190918195907-bd6ac527cfd2
go: finding k8s.io/apimachinery v0.0.0-20190817020851-f2f3a405f61d
go: finding k8s.io/code-generator v0.0.0-20190612205613-18da4a14b22b
go: finding github.com/globalsign/mgo v0.0.0-20181015135952-eeefdecb41b8
go: finding github.com/go-openapi/strfmt v0.17.0
go: finding github.com/go-openapi/spec v0.17.2
go: finding github.com/go-openapi/runtime v0.17.2
go: finding github.com/go-openapi/analysis v0.17.2
go: finding github.com/emicklei/go-restful
v0.0.0-20170410110728-ff4f55a20633
go: finding github.com/go-openapi/errors v0.17.2
go: finding github.com/prometheus/client_golang v0.9.2
go: finding k8s.io/apiserver v0.0.0-20190918200908-1e17798da8c1
go: finding github.com/go-openapi/validate v0.18.0
go: finding github.com/mailru/easyjson v0.0.0-20180823135443-60711f1a8329
go: finding github.com/go-openapi/errors v0.17.0
go: finding github.com/prometheus/common v0.0.0-20181126121408-4724e9255275
go: finding github.com/prometheus/procfs v0.0.0-20181204211112-1dc9a6cbc91a
go: finding github.com/asaskevich/govalidator
v0.0.0-20180720115003-f9ffefc3facf
go: finding github.com/matttproud/golang_protobuf_extensions v1.0.1
go: finding github.com/go-openapi/swag v0.17.0
go: finding golang.org/x/net v0.0.0-20181005035420-146acd28ed58
go: finding github.com/go-openapi/jsonreference v0.17.0
go: finding github.com/coreos/bbolt v1.3.1-coreos.6
go: finding golang.org/x/tools v0.0.0-20190328211700-ab21143f2384
go: finding github.com/pborman/uuid v1.2.0
go: finding github.com/prometheus/client_model
v0.0.0-20180712105110-5c3871d89910
go: finding github.com/coreos/pkg v0.0.0-20180108230652-97fdf19511ea
go: finding github.com/grpc-ecosystem/grpc-gateway v1.3.0
go: finding github.com/coreos/etcd v3.3.13+incompatible
go: finding github.com/go-openapi/jsonpointer v0.17.0
go: finding github.com/docker/docker v0.7.3-0.20190327010347-be7ac8be2ae0
go: finding gopkg.in/yaml.v1 v1.0.0-20140924161607-9f9df34309c0
go: finding github.com/go-openapi/validate v0.17.0
go: finding github.com/NYTimes/gziphandler
v0.0.0-20170623195520-56545f4a5d46
go: finding github.com/go-openapi/swag v0.17.2
go: finding github.com/globalsign/mgo v0.0.0-20180905125535-1ca0a4f7cbcb
go: finding gonum.org/v1/gonum v0.0.0-20190331200053-3d26580ed485
go: finding github.com/inconshreveable/mousetrap v1.0.0
go: finding go.uber.org/atomic v0.0.0-20181018215023-8dc6146f7569
go: finding github.com/PuerkitoBio/urlesc v0.0.0-20170810143723-de5bf2ad4578
go: finding github.com/munnerz/goautoneg v0.0.0-20120707110453-a547fc61f48d
go: finding github.com/jonboulle/clockwork
v0.0.0-20141017032234-72f9bd7c4e0c
go: finding github.com/go-openapi/jsonreference v0.19.0
go: finding github.com/xiang90/probing v0.0.0-20160813154853-07dd2e8dfe18
go: finding github.com/grpc-ecosystem/go-grpc-prometheus
v0.0.0-20170330212424-2500245aa611
go: finding github.com/spf13/cobra v0.0.0-20180319062004-c439c4fa0937
go: finding k8s.io/gengo v0.0.0-20190116091435-f8a0810f38af
go: finding gopkg.in/natefinch/lumberjack.v2
v2.0.0-20150622162204-20b71e5b60d7
go: finding google.golang.org/genproto v0.0.0-20170731182057-09f6ed296fc6
go: finding github.com/coreos/go-semver v0.0.0-20180108230905-e


Can you help?
Thank you,
Sushant

**Comment by archie951** [comment]
Please update the git version on both node and controller. I am suspecting that the current version will be 1.8.x. You need to update it to 2.1.x. For doing so please run below commands on both VM's:

yum remove -y git*
yum install -y --enablerepo=ius-archive git2u-all

Regards,
Archit

**Comment by sushantsj** [comment]
Hey archit,
Thanks a lot, it is working now and build successfully.

Thank you,
Sushant Jha

**Comment by archie951** [comment]
Oh that's great to hear Sushant. No problem at all.

@amr-mokhtar - Hi, Could you please close this ticket. 
Thanks,
Archit
